lib/hashie/clash.rb

```diff
@@ -54,7 +54,9 @@ def merge_store(key, *args) #:nodoc:
       case args.length
       when 1
         val = args.first
-        val = self[key].merge(val) if self[key].is_a?(::Hash) && val.is_a?(::Hash)
+        if self[key].is_a?(::Hash) && val.is_a?(::Hash)
```

Comment:
@dblock is not a change of one-liner, it also changes the value assigned to `val`.